### PR TITLE
Upgrade to Bootstrap 4.0.0.alpha6

### PR DESCRIPTION
**Pull request by jhawthorn**
Previously we were on Bootstrap 4.0.0.alpha2. Changes can be seen in their [Release notes](https://github.com/twbs/bootstrap/releases)

Notable to us:
* `col-xs-3` et al. are gone. Replaced with `col-3`.
* Flexbox is on by default :+1:  (we had already turned on the optional flexbox support)
* Classes for breadcrumbs changed (easy fix)
* Some variables and mixins changed

I can't be 100% sure this didn't mess up any styling, but in a quick browse through the admin I didn't spot anything off.

### CODE REVIEW

File: backend/app/assets/stylesheets/spree/backend/shared/_typography.scss [hunk 1]
```diff
@@ -1,5 +1,10 @@
 // Base
 //--------------------------------------------------------------
+
+html {
+  font-size: $font-size-root;
```
Comment (jhawthorn) [inline backend/app/assets/stylesheets/spree/backend/shared/_typography.scss]: I would like that, but I think we should do this in a separate PR. Removing this would move us from `13px` to `16px` (maybe), which is a really significant change, which I'm sure will cause some issues. I'd rather we look at that separately.

File: backend/vendor/assets/stylesheets/solidus_admin/bootstrap/_media.scss [hunk 2]
```diff
@@ -1,90 +1,8 @@
...
-
-
-//
-// Media list variation
-//
-
-.media-list {
-  padding-left: 0;
-  list-style: none;
+.media-body {
```
Comment (jhawthorn) [inline backend/vendor/assets/stylesheets/solidus_admin/bootstrap/_media.scss]: Right. I forgot this.

Bootstrap removed the `.media-left` class https://v4-alpha.getbootstrap.com/layout/media-object/ instead asking you to just specify `.mr-3`. :+1: Your padding solution is a good fix.

File: backend/app/assets/stylesheets/spree/backend/shared/_typography.scss (hunk 1 above)
Comment (tvdeyen) [inline backend/app/assets/stylesheets/spree/backend/shared/_typography.scss]: As long as we set `$font-size-base` to `$body-font-size` everything should be fine.

Comment (jhawthorn) [inline backend/app/assets/stylesheets/spree/backend/shared/_typography.scss]: That changes the size of our `rem`, which I don't want to do. It's awkward that our base font size doesn't match a `rem`, making `rem` pretty useless if it doesn't match up with anything else on the page. If we're setting `$font-size-base` to a `px` value anyways we're not improving accessibility in any way.

I appreciate the argument that using the browser's default font size is a good thing. But if we aren't (and I think it's too big a change for this PR) the way bootstrap used to do it, and the way we're doing it here, by setting a size on the root element, is best.

**Comment by jhawthorn** [comment]
@tvdeyen I've pulled in your changes other than changing the font size.

File: backend/app/assets/stylesheets/spree/backend/shared/_typography.scss (hunk 1 above)
Comment (tvdeyen) [inline backend/app/assets/stylesheets/spree/backend/shared/_typography.scss]: Agreed!
